bftengine/src/bftengine/messages/CheckpointMsg.cpp

```diff
@@ -11,31 +11,48 @@
 
 #include "CheckpointMsg.hpp"
 #include "assertUtils.hpp"
+#include "Crypto.hpp"
+#include "SigManager.hpp"
 
 namespace bftEngine {
 namespace impl {
 
-CheckpointMsg::CheckpointMsg(ReplicaId senderId,
+CheckpointMsg::CheckpointMsg(ReplicaId genReplica,
                              SeqNum seqNum,
                              const Digest& stateDigest,
                              bool stateIsStable,
                              const concordUtils::SpanContext& spanContext)
-    : MessageBase(senderId, MsgCode::Checkpoint, spanContext.data().size(), sizeof(Header)) {
+    : MessageBase(genReplica,
+                  MsgCode::Checkpoint,
+                  spanContext.data().size(),
+                  sizeof(Header) + SigManager::instance()->getMySigLength()) {
   b()->seqNum = seqNum;
   b()->stateDigest = stateDigest;
   b()->flags = 0;
+  b()->genReplicaId = genReplica;
   if (stateIsStable) b()->flags |= 0x1;
   std::memcpy(body() + sizeof(Header), spanContext.data().data(), spanContext.data().size());
 }
 
+void CheckpointMsg::sign() {
+  auto sigManager = SigManager::instance();
+  sigManager->sign(body(), sizeof(Header), body() + sizeof(Header) + spanContextSize(), sigManager->getMySigLength());
+}
+
 void CheckpointMsg::validate(const ReplicasInfo& repInfo) const {
   ConcordAssert(type() == MsgCode::Checkpoint);
   ConcordAssert(senderId() != repInfo.myId());
 
-  if (size() < sizeof(Header) + spanContextSize() || (!repInfo.isIdOfReplica(senderId())) ||
+  auto sigManager = SigManager::instance();
+  auto sigLen = sigManager->getMySigLength();
```

Comment:
You are not the one to sign, I think you should use sigManager::getSigLength

Reply:
Yes, definitely, will be fixed, thanks!